### ISSUE TITLE
Dashboards: Import Breaks Datasource Targets into String Array

### ISSUE THREAD

**Issue by swiffer** [grafana/grafana#77325]
### What happened?

On Import of a Dashboard created with Grafana 8.3.4 (https://github.com/adriankumpf/teslamate/blob/master/grafana/dashboards/statistics.json) into Grafana 10.2.0 something seems to break the model code resulting in a broken query editor

![grafik](https://github.com/grafana/grafana/assets/2990373/79aa5327-adec-46d3-85bf-b5c7d6282dec)

```json
  "panels": [
    {
      "collapsed": false,
      "datasource": "TeslaMate",
      "gridPos": {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 4,
      "panels": [],
      "repeat": "car_id",
      "targets": [
        {
          "datasource": {
            "0": "T",
            "1": "e",
            "2": "s",
            "3": "l",
            "4": "a",
            "5": "M",
            "6": "a",
            "7": "t",
            "8": "e"
          },
          "refId": "A"
        }
      ],
      "title": "$car_id",
      "type": "row"
    },
```

### What did you expect to happen?

The query editor should work just fine

### Did this work before?

yes

### How do we reproduce it?

1. Import Dashboard from here: https://github.com/adriankumpf/teslamate/blob/master/grafana/dashboards/statistics.json
2. Open the Panel config of the Table, see broken Query Editor
3. Check Json Model (Line 62+, Line 518+, Line 556+, Line 593+) - that looks broken

### Is the bug inside a dashboard panel?

Happens with different panels, therefore seems to be a generic / migration related issue

### Environment (with versions)?

Grafana: 10.2.0
OS: Win 11
Browser: Firefox

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

**Comment by tonypowa** [comment]
hi @swiffer 

the issue seems to be caused by the uid of the datasource, which has changed over the Grafana upgrades.

you would need to Replace old UID with the new datasource UID

Here is how you can [retrieve the UID](https://github.com/grafana/grafana/discussions/53038#discussioncomment-4701694) 

hope that solves the issue

**Comment by swiffer** [comment]
Not Sure, the original Dashboard has No uid in the json, and the Name is still TeslaMate.

**Comment by tonypowa** [comment]
@swiffer 
Just forwarded the issue to the dashboard squad for further review

**Comment by swiffer** [comment]
@tonypowa  - we fixed in in the teslamate repo by refering to another schemaVersion in the Dashboard json

https://github.com/teslamate-org/teslamate/pull/3454/files

It however would be great to see updated docs on how to provision data-sources & dashboards referring to datasources using the latest schema version here:

https://grafana.com/docs/grafana/latest/administration/provisioning/#config-file

**Comment by tonypowa** [comment]
Thank you for sharing the solution 

I tested provisioning a dashboard with Prometheus as data source, and despite the `schemaVersion` in the JSON was old, the dashboard was imported with the latest version available (38). 

I'm forwarding this issue to the dashboard squad to understand how it works and to explore what can be done to ensure the `schemaVersion` updates automatically when provisioning dashboards that use other data sources such as teslamate data source.

**Comment by ivanortegaalba** [comment]
Hi, I tried to reproduce this issue, and this is solved in the latest version of Grafana (10.4), so I'll close this issue. We're trying to limit open GitHub issues in order to better track planned work and features.

Thank you for taking the time to create this issue! Please let us know if it happens again!
